#!/usr/bin/env python

PACKAGE = "hero_chassis_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#add(name,type, level,description, default, min,max)
gen.add("wheel_track", double_t, 0, "linear velocity of the robot",0.5,-1.0,1.0)
gen.add("wheel_base", double_t, 0, "rotation velocity of the robot", 0.475 ,-1.0,1.0)
size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")
gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)
exit(gen.generate(PACKAGE,"hero_chassis_controller","tutorial"))
